### PR TITLE
Issue #3295894 by navneet0693: Junk data crashes queues.

### PR DESCRIPTION
**Pull request by navneet0693**
## Problems

**Problem 1**
This was a special case scenario where a queue was created by activity_creator_logger . It was trying to send notifications/emails to all the members.

`Error: Call to a member function id() on null in Drupal\activity_basics\Plugin\ActivityContext\ContentInMyGroupActivityContext->getRecipients() (line 160 of html/profiles/contrib/social/modules/custom/activity_basics/src/Plugin/ActivityContext/ContentInMyGroupActivityContext.php) #0 html/profiles/contrib/social/modules/custom/activity_creator/src/Plugin/QueueWorker/ActivityWorkerLogger.php(48): Drupal\activity_basics\Plugin\ActivityContext\ContentInMyGroupActivityContext->getRecipients(Array, 0, 0)`

While processing all the memberships at https://github.com/goalgorilla/open_social/blob/11.1.13/modules/custom/activity_basics/src/Plugin/ActivityContext/ContentInMyGroupActivityContext.php#L157

$membership->getUser() was returning NULL when it was trying to fetch the user with a particular user ID.

This particular user was not existing in the system.

**Problem 2**

When content is reported (technically flagged) an activity is created to send out notifications. Sometimes, the flagging entity doesn't exist, which then crashes the queue.

` [error]  Error: Call to a member function getFlaggable() on null in social_content_report_tokens_alter() (line 111 of /app/html/profiles/contrib/social/modules/social_features/social_content_report/social_content_report.tokens.inc`

## Solution
To avoid such scenarios, 

1. Add a check if the $membership->getUser() returns a valid user entity.
2. Add a check if $flaggable is an existing entity.

## Issue tracker
https://www.drupal.org/project/social/issues/3295894

## Theme issue tracker
N.A

## How to test
**For Problem 1**
- [ ] It's hard to reproduce this scenario but one can try.
- [ ] Create a group
- [ ] Add a certain user to that group
- [ ] Create a content in the user group
- [ ] Delete the user data from `user` and `user_field_data` tables ( or try deleting via UI by checking "keep the content" option.)
- [ ] Run cron

**For Problem 2**
- [ ] It's kind of hard to reproduce.

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
N.A

## Release notes
Our queues get stuck once a user from the platform was deleted but its membership from a certain group where it was a membership didn't get removed. We resolved these issues by adding defensive checks to our code.

## Change Record
N.A

## Translations
N.A

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3036-3ewnvldq1dwpdljj6rpjfug6ngg1ods0.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62cfb9eebbf0e06f92099086

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3036-3ewnvldq1dwpdljj6rpjfug6ngg1ods0.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62cfb9eebbf0e06f92099086

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3036-3ewnvldq1dwpdljj6rpjfug6ngg1ods0.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62cfb9eebbf0e06f92099086

File: modules/custom/activity_basics/src/Plugin/ActivityContext/ContentInMyGroupActivityContext.php [hunk 1]
```diff
@@ -157,7 +157,16 @@ public function getRecipients(array $data, $last_uid, $limit) {
         foreach ($memberships as $membership) {
           // Check if this not the created user and didn't mute the group
           // notifications.
```
[the remark this answers is not in the document]
Comment (navneet0693) [reply modules/custom/activity_basics/src/Plugin/ActivityContext/ContentInMyGroupActivityContext.php:159]: This change will be unrelated to this PR.

[the remark this answers is not in the document]
Comment (Ressinel) [reply modules/custom/activity_basics/src/Plugin/ActivityContext/ContentInMyGroupActivityContext.php:159]: @navneet0693 There is a duplicate of this comment. For now in your branch you can see:
```
// Check if this not the created user and didn't mute the group
// notifications.
// Check if this is not the created user and didn't mute the group
// notifications.
```

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3036-3ewnvldq1dwpdljj6rpjfug6ngg1ods0.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62cfb9eebbf0e06f92099086

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3036-3ewnvldq1dwpdljj6rpjfug6ngg1ods0.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62cfb9eebbf0e06f92099086

**Review by Ressinel — APPROVED** [review]
Looks good to me.

**Comment by navneet0693** [comment]
🍒 - picked to 11.4.x, 11.3.x, 11.2.x and 11.1.x

**Pull request merged**